funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_worker.py

```diff
@@ -135,17 +134,21 @@ def handler(self, signum, frame):
     def _send_registration_message(self):
         log.debug("Sending registration")
         payload = {"worker_id": self.worker_id, "worker_type": self.worker_type}
-        self.task_socket.send_multipart([b"REGISTER", pickle.dumps(payload)])
+        self.task_socket.send_multipart([b"REGISTER", dill.dumps(payload)])
 
     def start(self):
         log.info("Starting worker")
-        self._send_registration_message()
+
+        result = self.registration_message()
+        task_type = b"REGISTER"
+        log.debug("Sending registration")
+        self.task_socket.send_multipart([task_type, dill.dumps(result)])  # Byte encoded
```

Comment:
This change looks unrelated, unless I'm missing something.
Was this maybe a change made while testing which slipped in here?

I think this is better to revert than accept, but I'm happy to be convinced otherwise if there's a reason to change it.

Reply:
You are right that it was a change reverted during merge that is unrelated.  Putting it back the way it is in main.